WaybackWhen.py

```
@@ -970,30 +984,66 @@ def run(self):
+        try:
+            crawler_executor = concurrent.futures.ThreadPoolExecutor(max_workers=self.max_crawler_workers)
+            archiver_executor = concurrent.futures.ThreadPoolExecutor(max_workers=self.max_archiver_workers)
+
+            while True:
+                current_time = time.time()
+
+                # Check for max crawl runtime
+                if crawling_enabled and SETTINGS["max_crawl_runtime"] > 0 and (current_time - crawl_process_start_time) > SETTINGS["max_crawl_runtime"]:
+                    log_message("INFO", f"Max crawling runtime of {SETTINGS['max_crawl_runtime']} seconds reached. Stopping new crawl tasks and cancelling active ones.", debug_only=False)
+                    crawling_enabled = False
+                    for future, url, root_domain in list(self.crawling_futures_set): # Iterate over a copy
```

Ruff (0.14.14)

[warning] Loop control variable `root_domain` not used within loop body

Rename unused `root_domain` to `_root_domain`

(B007)